### ISSUE TITLE
Use of undefined resources or methods gets stuck

### ISSUE THREAD

**Issue by gonenduk** [cdimascio/express-openapi-validator#90]
Suppose we define a GET method on the resource /api/users.
If we try to access GET /api/users2 - we never get a reply instead of 404 (resource not found).
If we try to access POST /api/users - we never get a reply instead of 405 (method not allowed).

Some people might want to allow a pass through if resource is not found or method not defined. Maybe an option should be available to allow it (defaults to off) - but the process should not hang.

**Comment by comino** [comment]
I think the behaviour you request describe is the expected behaviour. 

From openapi.security.js: 
```
   if (!req.openapi) {
            // this path was not found in open api and
            // this path is not defined under an openapi base path
            // skip it
            return next();
        }
        // const securities: OpenAPIV3.SecurityRequirementObject[] =
        //   req.openapi.schema.security;
        // use the local security object or fallbac to api doc's security or undefined
        const securities = req.openapi.schema.security || context.apiDoc.security;
```

The problem is, that the check for undefined paths is falling through in my tests and throwing at "req.openapi.schema.security" leading to a unresponsive behavior. 

Send: https://github.com/cdimascio/express-openapi-validator/pull/91

**Comment by gonenduk** [comment]
cool!
And what do you get if you call a defined url but with an undefined method?

**Comment by comino** [comment]
404 / 405 works in unit test and sample project with the simple one-line-fix.

The issue can be reproduced by using a top level security definiton. The existing unit test only tested this for per endpoint security definition.

**Comment by cdimascio** [comment]
fixed in v2.12.2. Thanks @comino!

**Comment by gonenduk** [comment]
Verified the fix works on my scenario. It does :)
